Frameworks/Foundation/NSHTTPURLResponse.mm

```diff
@@ -147,29 +153,68 @@ - (id)copyWithZone:(NSZone*)zone {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (BOOL)supportsSecureCoding {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return YES;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-- (id)initWithCoder:(NSCoder*)decoder {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (void)encodeWithCoder:(NSCoder*)coder {
+    [coder encodeInteger:_statusCode forKey:@"NS.statusCode"];
+    [coder encodeObject:_allHeaderFields forKey:@"NS.allHeaderFields"];
+    [coder encodeObject:_HTTPVersion forKey:@"NS.HTTPVersion"];
+    [super encodeWithCoder:coder];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-- (void)encodeWithCoder:(NSCoder*)coder {
-    UNIMPLEMENTED();
+- (instancetype)initWithCoder:(NSCoder*)coder {
+    if (self = [super initWithCoder:coder]) {
+        _statusCode = [coder decodeIntegerForKey:@"NS.statusCode"];
+        NSDictionary* headerFields = [coder decodeObjectOfClass:[NSMutableDictionary class] forKey:@"NS.allHeaderFields"];
+		if (headerFields != nil) {
+			_allHeaderFields = _constructCaseInsensitiveDictionary(headerFields);
+	    }
+
+        _HTTPVersion = [coder decodeObjectOfClass:[NSString class] forKey:@"NS.HTTPVersion"];
+    }
+
+    return self;
+}
+
+/**
+ @Status Interoperable
+*/
+- (BOOL)isEqual:(id)other {
+    if ((other == nil) || ![other isKindOfClass:[NSHTTPURLResponse class]]) {
+        return NO;
+    }
+
+	if (other == self) {
+		return YES;
+	}
+
```

Comment:
This could move up to be the first check to save even more;
Also, tab characters!